nexus/src/db/collection_attach.rs

```diff
@@ -338,9 +332,10 @@ where
     AttachToCollectionStatement<ResourceType, V, C>: Send,
 {
     /// Issues the CTE asynchronously and parses the result.
-    pub async fn attach_and_get_result_async(
+    pub async fn attach_and_get_result_async<ConnErr>(
         self,
-        pool: &bb8::Pool<ConnectionManager<DbConnection>>,
+        conn: &(impl async_bb8_diesel::AsyncConnection<DbConnection, ConnErr>
+              + Sync),
```

Comment:
Basically all of our non-transaction code acts on a "bb8::Pool". However, the `transaction_async` method expects to act on an `async_bb8_diesel::Connection`.

Fortunately, `async_bb8_diesel::AsyncConnection` is implemented on *both* objects, so if we operate on that interface, we can interop with *either* the pool *or* a connection object in a transaction context. This pattern of `impl AsyncConnection` is used in a few spots in this patch, and is how I expect we could make code "generic to being run in a txn" in the future.

Unfortunately, the errors are different - we only propagate "pool-related errors" if you're actually acting on a pool - so that is made generic here too (see: `ConnErr`). If being generic over errors is only a pain, and not particularly useful, we could consider just propagating a `PoolError` everywhere (it's a superset of the other errors that could be returned from `async_bb8_diesel`).